src/Npgsql/NpgsqlDatabaseInfo.cs

```diff
@@ -231,6 +231,7 @@ public static void RegisterFactory(INpgsqlDatabaseInfoFactory factory)
 
         internal static async Task<NpgsqlDatabaseInfo> Load(NpgsqlConnection conn, NpgsqlTimeout timeout, bool async)
         {
+            // Foreach for array in translated into for, so we can safely change one array to another
```

Comment:
Not needed, it's a very obvious thing (:

Reply:
Are you saying that a foreach implicitly take a local copy of the reference to the array? i.e. if we swap the Factories volatile reference while iteration is happening, do we continue to iterate on the old version (as we should)?

Reply:
```csharp
foreach (var factory in Factories)
// is translated to
var factories = Factories;
for (var i = 0; i < factories.Count; i++)
```

It happens even if `Factories` isn't volatile.

Reply:
Thanks, I wasn't clear on that...!